torch/distributions/kl.py

```diff
@@ -278,6 +308,21 @@ def _kl_laplace_laplace(p, q):
     return t1 + t2 + t3 - 1
 
 
+@register_kl(MultivariateNormal, MultivariateNormal)
+def _kl_multivariatenormal_multivariatenormal(p, q):
+    # From https://en.wikipedia.org/wiki/Multivariate_normal_distribution#Kullback%E2%80%93Leibler_divergence
+    if p.event_shape != q.event_shape:
+        raise ValueError("KL-divergence between two Multivariate Normals with\
+                          different event shapes cannot be computed")
+
+    term1 = _batch_diag(q.scale_tril).log().sum(-1) - _batch_diag(p.scale_tril).log().sum(-1)
+
+    term2 = _batch_trace(_batch_mm(_batch_inverse(q.covariance_matrix), p.covariance_matrix))
+
+    term3 = _batch_mahalanobis(q.scale_tril, (q.loc - p.loc))
+    return term1 + 0.5 * (term2 + term3 - p.event_shape[0])
```

Comment:
I guess here we should have `0.5 * term1` ?

Reply:
I thought the trace would involve fewer computations than the sum. Regarding `0.5 * term1`, @fritzo and I have discussed it in the `probtorch/pytorch` https://github.com/probtorch/pytorch/pull/144.

Reply:
Thank you @vishwakftw ! Got it ^^